qutebrowser/browser/network/networkmanager.py

```diff
@@ -358,6 +358,23 @@ def createRequest(self, op, req, outgoing_data):
             dnt = '0'.encode('ascii')
         req.setRawHeader('DNT'.encode('ascii'), dnt)
         req.setRawHeader('X-Do-Not-Track'.encode('ascii'), dnt)
+
+        def same_domain():
+            tabbed_browser = objreg.get('tabbed-browser', scope='window', window=self._win_id)
+            view = tabbed_browser.currentWidget()
+            print(req.url().host() == view.url().host(), repr(view.url().host()), repr(req.url().host()))
+            # TODO: We probably want to allow headers if we're on test.com and
+            # doing a request to www.test.com? Or maybe a new settings value for
+            # this?
+            return req.url().host() == view.url().host()
```

Comment:
It's not just new TLD's, also:

	*.nom.br
	abruzzo.it
	compute.amazonaws.com

etc.

I would be in favour of shipping with our own list. We can use the `tldextract` code, which can update the list as well (so even people running very outdated versions of Qt/qutebrowser should be okay).

It's BSD-licensed, so we can jist ship qutebrowser with it (ie. no need for extra install dependency).